week1/homework/src/server.js

```diff
@@ -5,16 +5,42 @@ const http = require('http');
 /* `createServer` MUST return an instance of `http.Server` otherwise the tests
  * will fail.
  */
-function createServer(port) {
+function createServer() {
   let state = 10;
-
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+    switch (request.url) {
+      case '/state':
+        response.setHeader('Content-Type', 'application/json');
+        const stringifiedState = JSON.stringify({ state: state });
+        response.write(stringifiedState);
+        break;
+      case '/add':
+        response.setHeader('Content-Type', 'application/json');
+        const stringifiedState2 = JSON.stringify({ state: ++state });
```

Comment:
PRO TIP: It is a good idea to separate your app logic (addition) from technical implementation details (stringify)

You had the right idea with with leaving response.write in a separate line 👍  I think the same can be done with stringify

```
++state;
JSON.stringify({state: state});
```